test: update notification e2e tests

Prithpal-Sooriya opened the pull request:
## **Description**

Updates notification e2e spec to handle clicking and viewing all the different notification items we support.

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMASSETS-837

## **Manual testing steps**

N/A - this adds e2e test coverage.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

Prithpal-Sooriya committed:
test: update notification e2e tests

Prithpal-Sooriya committed:
test: fix tests - scrolling to items

Prithpal-Sooriya committed:
test: update snapshot

Prithpal-Sooriya committed:
test: add correct test scrolling

Prithpal-Sooriya commented on app/components/UI/Notification/List/styles.ts:
```diff
@@ -94,7 +94,7 @@ export const createStyles = ({ colors, typography }: Theme) =>
       flex: 1,
     },
-    list: { flexGrow: 1 },
+    list: { flexGrow: 1, paddingBottom: 100 },
```
Legit UI bug. If you have a lot of notifications, it is hard to select the item at the bottom since the FAB to mark as read is blocking the last content.

Now we have some padding/buffer at the bottom to allow additional scroll area for the last item to be in view and clickable.

Prithpal-Sooriya commented on e2e/pages/Notifications/NotificationMenuView.js:
```diff
@@ -23,12 +23,19 @@ class EnableNotificationsModal {
     id,
   ) {
-    return Matchers.getElementByID(NotificationMenuViewSelectorsIDs.ITEM(id));
+    return /** @type {Promise<Detox.IndexableNativeElement>} */ (
+      Matchers.getElementByID(NotificationMenuViewSelectorsIDs.ITEM(id))
+    );
```
Using `@ts-check` and JSDOC types for some strict types for JS.
There was a type-mismatch, so used a JSDoc type assertion here.

Prithpal-Sooriya commented on e2e/specs/notifications/utils/mocks.js:
```diff
@@ -55,18 +55,23 @@ mockListNotificationsResponse.response = [
   createMockNotificationLidoWithdrawalCompleted(),
-];
+].map((n, i) => {
+  const date = new Date();
+  date.setDate(date.getDate() - i);
+  n.created_at = date.toString();
+  return n;
```
Ordering of notifications to make it easier to use detox to select through each of the items.

Prithpal-Sooriya committed:
refactor: update scroll down amount to make sure items are visible

hjetpoluru commented on e2e/specs/notifications/enable-notifications-after-onboarding.spec.js:
```diff
@@ -92,10 +85,12 @@ describe(SmokeNotifications('Notification Onboarding'), () => {
     await NotificationDetailsView.tapOnBackButton();
 
     // Wallet Annonucement Details
```
```suggestion
    // Wallet Announcement Details
```

hjetpoluru reviewed (dismissed):
LGTM

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into test/update-notification-e2e-spec

Prithpal-Sooriya committed:
test: fix e2e on android

android had issues with scrolling through items due to the FAB.
So we have made most mock notifications read. By the time it needs to scroll, all notifications will have been read and the FAB disappears

Prithpal-Sooriya commented on e2e/specs/notifications/utils/mocks.js:
```diff
@@ -55,18 +55,24 @@ mockListNotificationsResponse.response = [
+  const date = new Date();
+  date.setDate(date.getDate() - i);
+  n.created_at = date.toString();
+  n.unread = false;
```
Most mock notifications have been marked as read.
This is because mobile uses a FAB for marking all notifications as read. (see image)
![Screenshot 2025-05-20 at 14 26 06](https://github.com/user-attachments/assets/beaab462-11dc-40ba-80b3-3887ceb843f1)


This FAB blocks causes issues on android QA tests, so to unblock the tests, some notifications will be already read. This way, when it comes time to scroll the page, there is nothing interfering.
